interfaces/apparmor/template.go

```diff
@@ -1001,6 +1001,12 @@ profile snap-update-ns.###SNAP_INSTANCE_NAME### (attach_disconnected) {
   /run/snapd/lock/###SNAP_INSTANCE_NAME###.lock rwk,
   /run/snapd/lock/.lock rwk,
 
+  # While the base abstraction has rules for encryptfs encrypted home and
+  # private directories, it is missing rules for directory read on the toplevel
+  # directory of the mount (LP: #1848919)
+  owner @{HOME}/.Private/ r,
+  owner @{HOMEDIRS}/.ecryptfs/*/.Private/ r,
```

Comment:
Why the use of HOMEDIRS here rather than HOME?

Reply:
I it my understanding that ecryptfs keeps those directories directly under /home, i.e. `/home/.ecryptfs/<user>`. So homedirs expands to what we want `{/home,/root/,/other/homes}`, but `@{HOME}` includes the user directory part, so it expands to `{/home/*/,...}`, there's a similar pattern in the common app profile https://github.com/snapcore/snapd/blob/3cfa28a0fc09b476e911b8cc103792e3577486fc/interfaces/apparmor/template.go#L111

Reply:
ah yep of course - thanks @bboozzoo